### DISCUSSION TITLE
Issues installing eth-brownie with python 3.11 and windows 11

### DISCUSSION THREAD

**Discussion by mattjaf** [smartcontractkit/full-blockchain-solidity-course-py#1799] [Q&A] [answered]
I even got errors when installing with pip...

```sh
PS C:\Users\mattj> pipx install eth-brownie       
Fatal error from pip prevented installation. Full pip output in file:
    C:\Users\mattj\.local\pipx\logs\cmd_2022-12-01_19.55.25_pip_errors.log

pip failed to build packages:
    bitarray
    cytoolz
    yarl

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.34.31933\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
    cytoolz/dicttoolz.c(209): fatal error C1083: Cannot open include file: 'longintrepr.h': No such file or directory
    yarl/_quoting_c.c(196): fatal error C1083: Cannot open include file: 'longintrepr.h': No such file or directory

Error installing eth-brownie.
```

**Comment by cromewar** [comment]
Hello @mattjaf  as you mentioned on you other post this could be solved installed cytools trough Visual Studio.
Cheers.

**Reply by mattjaf** [reply]
That's like one of the first things I tried and it didnt seem to fix it.

I've tried installing, bitarray, cytoolz, yarl.
Downgrading ganache to 7.0.2 with node 16.
Tried with pip, instead of pip x

And few other things

**Comment by mattjaf** [comment]
Wait is there something else I was suppose to select in the visual studio installer?

I tried installing cytoolz through pip

**Reply by cromewar** [reply]
in visual studio you select the python development tools that should be enough, also another solution is using WSL (windows subsystem for linux) and manage your dependencies there.

**Reply by mattjaf** [reply]
i imagine the python development tools is included in the c++/cli build tools,

i have wsl, although i was planning on setting up my global environment in powershell, if that the right approach...

It might be a good idea for me to set up a python virtual environment,

do you have experience with venv?

**Reply by cromewar** [reply]
> i imagine the python development tools is included in the c++/cli build tools,

Not really be sure to install python tools as well.

> i have wsl, although i was planning on setting up my global environment in powershell, if that the right approach...

Depends a lot on your preferences, but any experienced developer will tell you that windows is a mess on library management so they are used to use Linux or Mac (and WSL to have linux inside windows lol).

> do you have experience with venv?

You can use venv, virtualenv or conda environments any of those should work. Virtualenv is one of the easiest to use.

**Comment by mattjaf** [comment, marked as answer]
installed fine on python 3.9 might not be compatible with 3.11 breaking changes

**Reply by cromewar** [reply]
The best version for smart contract development is 3.8 :)

**Reply by mattjaf** [reply]
i used 3.8 on windows 7 for a bout a year, it worked nice
